src/evict/evict_page.c

```diff
@@ -512,15 +546,20 @@ __evict_child_check(WT_SESSION_IMPL *session, WT_REF *parent)
              *     3. If we do not but we're an eviction thread, go ahead. We will get a snapshot
              *        shortly and any committed operation will be visible in it.
              *     4. Otherwise, check if the operation is globally visible.
+             *
+             * Even though we specifically can't evict prepared truncations, we don't need to deploy
+             * the special-case logic for prepared transactions in __wt_page_del_visible; prepared
+             * transactions aren't committed so they'll fail the first check.
              */
-            if (!__wt_page_del_committed(child->ft_info.del))
+            if (!__wt_page_del_committed(child->page_del))
                 visible = false;
             else if (F_ISSET(session->txn, WT_TXN_HAS_SNAPSHOT))
-                visible = __wt_page_del_visible(session, child->ft_info.del, false);
+                visible = __wt_page_del_visible(session, child->page_del, false);
             else if (F_ISSET(session, WT_SESSION_EVICTION))
                 visible = true;
             else
-                visible = __wt_page_del_visible(session, child->ft_info.del, true);
+                visible = __wt_page_del_visible_all(session, child->page_del, false);
+            /* FUTURE: is there a reason this doesn't use WT_REF_UNLOCK? */
```

Comment:
I don't understand the comment here.

Reply:
There's a WT_REF_UNLOCK macro. It's not used here. It's not obvious if that was intentional or an oversight. It's not the same as the code here. (WT_REF_UNLOCK updates the ref hist and uses WT_PUBLISH, this does neither.) I didn't want to change it arbitrarily, but did want to leave a note for someone to think about it later. I wrote WT-9780 so this is also in its scope.